api/types/discoveryconfig/discoveryconfig.go

```diff
@@ -159,6 +153,13 @@ func (a *DiscoveryConfig) CheckAndSetDefaults() error {
 	return nil
 }
 
+// Clone returns a copy of the config.
+func (a *DiscoveryConfig) Clone() *DiscoveryConfig {
+	var copy *DiscoveryConfig
+	utils.StrictObjectToStruct(a, &copy)
+	return copy
```

Comment:
As we've discovered in the previous version of this PR, this clone is incompatible with the IsEqual method, which isn't great. Can we revert the change?